sql/opt/xform: TestExternal failed

sql/opt/xform.TestExternal [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10131164?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10131164?buildTab=artifacts#/) on master @ [bba88ee56adb765bdb99940d87aac3e1e62b2b74](https://github.com/cockroachdb/cockroach/commits/bba88ee56adb765bdb99940d87aac3e1e62b2b74):

Fatal error:

```
panic: test timed out after 55s

```
Stack: 

```
goroutine 201 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
              lt_s_symb = s_symb;
        ----
        project
         ├── columns: s_ex_id:5!null s_name:4!null co_name:21!null lt_price:43!null tt_is_mrkt:40!null tt_is_sell:39!null
         ├── cardinality: [0 - 1]
         ├── immutable
         ├── key: ()
         ├── fd: ()-->(4,5,21,39,40,43)
         ├── inner-join (cross)
         │    ├── columns: s_symb:1!null s_name:4!null s_ex_id:5!null s_co_id:6!null co_id:19!null co_name:21!null lt_s_symb:30!null last_trade.lt_price:32!null tt_id:37!null tt_is_sell:39!null tt_is_mrkt:40!null
         │    ├── cardinality: [0 - 1]
         │    ├── multiplicity: left-rows(zero-or-one), right-rows(zero-or-one)
         │    ├── key: ()
         │    ├── fd: ()-->(1,4-6,19,21,30,32,37,39,40), (6)==(19), (19)==(6)
         │    ├── inner-join (lookup company)
         │    │    ├── columns: s_symb:1!null s_name:4!null s_ex_id:5!null s_co_id:6!null co_id:19!null co_name:21!null
         │    │    ├── key columns: [6] = [19]
         │    │    ├── lookup columns are key
         │    │    ├── cardinality: [0 - 1]
         │    │    ├── key: ()
         │    │    ├── fd: ()-->(1,4-6,19,21), (19)==(6), (6)==(19)
         │    │    ├── scan security
         │    │    │    ├── columns: s_symb:1!null s_name:4!null s_ex_id:5!null s_co_id:6!null
         │    │    │    ├── constraint: /1: [/'ROACH' - /'ROACH']
         │    │    │    ├── cardinality: [0 - 1]
         │    │    │    ├── key: ()
         │    │    │    └── fd: ()-->(1,4-6)
         │    │    └── filters (true)
         │    ├── inner-join (cross)
         │    │    ├── columns: lt_s_symb:30!null last_trade.lt_price:32!null tt_id:37!null tt_is_sell:39!null tt_is_mrkt:40!null
         │    │    ├── cardinality: [0 - 1]
         │    │    ├── multiplicity: left-rows(zero-or-one), right-rows(zero-or-one)
         │    │    ├── key: ()
         │    │    ├── fd: ()-->(30,32,37,39,40)
         │    │    ├── scan last_trade
         │    │    │    ├── columns: lt_s_symb:30!null last_trade.lt_price:32!null
         │    │    │    ├── constraint: /30: [/'ROACH' - /'ROACH']
         │    │    │    ├── cardinality: [0 - 1]
         │    │    │    ├── key: ()
         │    │    │    └── fd: ()-->(30,32)
         │    │    ├── scan trade_type
         │    │    │    ├── columns: tt_id:37!null tt_is_sell:39!null tt_is_mrkt:40!null
         │    │    │    ├── constraint: /37: [/'TLS' - /'TLS']
         │    │    │    ├── cardinality: [0 - 1]
         │    │    │    ├── key: ()
         │    │    │    └── fd: ()-->(37,39,40)
         │    │    └── filters (true)
         │    └── filters (true)
         └── projections
              └── last_trade.lt_price:32::FLOAT8 [as=lt_price:43, outer=(32), immutable]
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExternal.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-28047

--- a/pkg/sql/opt/xform/BUILD.bazel
+++ b/pkg/sql/opt/xform/BUILD.bazel
@@ -61,7 +61,7 @@ go_library(
 
 go_test(
     name = "xform_test",
-    size = "small",
+    size = "medium",
     srcs = [
         "coster_test.go",
         "general_funcs_test.go",
@@ -70,7 +70,7 @@ go_test(
         "optimizer_test.go",
         "physical_props_test.go",
     ],
-    args = ["-test.timeout=55s"],
+    args = ["-test.timeout=295s"],
     data = glob(["testdata/**"]) + [
         "//c-deps:libgeos",
         "//pkg/sql/opt/testutils/opttester:testfixtures",

